### PR TITLE
STM32: Add the OpenMV-H7 R1 board

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds basic pin and clock definitions for the OpenMV-H7 board. The OpenMV-H7 is a development board for embedded machine vision using the STM32H743 processor. More information about the project can be found on on the website, [OpenMV.io](https://openmv.io/). 

This PR is naturally a little anemic for the scope of the OpenMV project, since it doesn't support the board's primary feature, the OV7725 Camera. OpenMV as supported with Micropython is an expansive project that actually uses a modified [fork of micropython](https://github.com/openmv/micropython/tree/abc992e9401f98829e07465c3836601b07358a37) as a submodule of their [source code](https://github.com/openmv/openmv). Including full support for the OpenMV in Circuitpython would probably require some strategy discussion with the OpenMV team and an intimidating amount of work.

That said, it's fun to think about what could be done with Circuitpython if it were to take full advantage of powerful chips like the H7 series. OpenMV not only does real time machine vision but even loops in STM32Cube.AI to enable optimized neural network use. Most computer vision on SBCs is at least interfaced with python, and it'd be cool to expose that on microcontrollers to more inexperienced users.

Board was tested with the LEDs, I2C (DRV2605), SPI(BMP), and UART (self-detect) with no issues. @tannewt @ladyada requesting a valid PID when you have a minute.

### CODE REVIEW

**Comment by ladyada** [comment]
nice! we do want to have some x-platform image capture soon - samd51 and imx rt also have camera capture peripherals :)

@kwagyeman take a look!

**Comment by kwagyeman** [comment]
Cool stuff.

We are about to bring an interface library online for the OpenMV Cam H7. I spent an extensive amount of time fixing PYB peripherals. So, you might only want to change the core logic.

File: ports/stm/boards/openmv_h7/pins.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    { MP_ROM_QSTR(MP_QSTR_P4), MP_ROM_PTR(&pin_PB10) },
+    { MP_ROM_QSTR(MP_QSTR_P5), MP_ROM_PTR(&pin_PB11) },
+    { MP_ROM_QSTR(MP_QSTR_P6), MP_ROM_PTR(&pin_PA05) },
+    { MP_ROM_QSTR(MP_QSTR_P7), MP_ROM_PTR(&pin_PD12) },
+    { MP_ROM_QSTR(MP_QSTR_P8), MP_ROM_PTR(&pin_PD13) },
+    { MP_ROM_QSTR(MP_QSTR_P9), MP_ROM_PTR(&pin_PD14) },
+    { MP_ROM_QSTR(MP_QSTR_LED1), MP_ROM_PTR(&pin_PC00) }, // RED
+    { MP_ROM_QSTR(MP_QSTR_LED2), MP_ROM_PTR(&pin_PC01) }, // GREEN
+    { MP_ROM_QSTR(MP_QSTR_LED3), MP_ROM_PTR(&pin_PC02) }, // BLUE
+    { MP_ROM_QSTR(MP_QSTR_LED4), MP_ROM_PTR(&pin_PE02) }, // IR
```
Comment (tannewt) [inline ports/stm/boards/openmv_h7/pins.c:17]: Why not name these after the color?

Comment (tannewt) [reply ports/stm/boards/openmv_h7/pins.c:17]: You can include two mappings so both names work as well.

**Comment by tannewt** [comment]
> We are about to bring an interface library online for the OpenMV Cam H7. I spent an extensive amount of time fixing PYB peripherals. So, you might only want to change the core logic.

What do you mean by interface library? I'd love to know more details.

File: ports/stm/boards/openmv_h7/pins.c (hunk 1 above)
Comment (hierophect) [reply ports/stm/boards/openmv_h7/pins.c:17]: Consistency, a lot of other boards go by number, as does the Micropython implementation of this one. I'll add the colors as an added option.

**Comment by kwagyeman** [comment]
I'm finishing up the process of releasing a high performance RPC library for the OpenMV Cam. This lets you use all the interfaces on the board to move large data structures (images or whatnot) at high data rates. For example, the library lets you call a function on the camera, pass a unlimited length (e.g. 2^32-1) byte struct to the camera, and then get an unlimited length byte struct result.

For example, I have SPI transferring RAW QQVGA RGB565 images between one OpenMV Cam and another at 20 FPS over a 20 MHz SPI bus link. The interface is stable and automatically recovers if you unplug any of the wires and then plug it back in. The SPI interface can run at 40 MHz or 80 MHz if the wire lengths are short enough (80 MHz would require two devices to share the same PCB).

I also have I2C working at 1 Mb/s, CAN operating at 1 Mb/s, USB VCP operating at 12 Mb/s, UART operating at 7.5 Mb/s, and WiFi operating at 10-12 Mb/s.

Each of these interfaces is self-healing so that if you disconnect and then reconnect on the interface while things are running there are no crashes. Higher level software just gets timeouts and then things start running again once the wire is put back in.

While writing this library I encountered terrible bugs with every interface and in the PYB module. That said, once the RPC library is out no-one will ever need to worry about how to get any bus working on the OpenMV Cam anymore.

Anyway, the library will also be available on the Arduino, RaspberryPi, and in python on the PC. We will also ship examples for all these platforms.

**Comment by hierophect** [comment]
@tannewt led alternatives added

**Comment by hierophect** [comment]
@tannewt wait this needed a valid PID

**Comment by tannewt** [comment]
Please make another PR:
```
OpenMV-H7 R1          VID 0x239A    PID 0x00A3  # bootloader
                                    PID 0x80A3  # arduino
                                    PID 0x80A4  # circuitpython
```

We should probably use draft for PRs that shouldn't be merged.

**Comment by hierophect** [comment]
> We should probably use draft for PRs that shouldn't be merged.

Yes, this was my bad. I'll use a draft PR next time I need a PID before merging.

**Pull request merged**